### PR TITLE
create-order api created

### PR DESCRIPTION
**Pull request by Mhshohag2256**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -25,9 +23,8 @@ app.get('/', (req, res) => res.json({ msg: 'Welcome to Sylica REST API' }));
 // define router paths
 app.use('/api/auth', authRouter);
 app.use('/api/products', productsRouter);
-app.use('/api/checkout', checkoutRouter)
-
-
+app.use('/api/checkout', checkoutRouter);
+app.use('/api/orders',authenticateToken,orderRouter);
```
Comment (zealptl) [inline src/server.js:27]: authenticateToken goes here `router.post('/create-order', createOrder)`, not where you use the route

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
 	},
 	"devDependencies": {
 		"@babel/plugin-transform-modules-commonjs": "^7.15.4",
+		"@types/jsonwebtoken": "^8.5.5",
```
Comment (zealptl) [inline package.json:45]: why did you installed this?

File: src/routes/controllers/ordersController/createOrder.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import order from '../../../models/order';
+import OrderModel from '../../../models/order';
+
+const createOrder = async (req, res) => {
+  try {
+    const user = req.user._id;
+
+    const { shippingAddress, billingAddress, totalAmount, productsPurchased } =
```
Comment (zealptl) [inline src/routes/controllers/ordersController/createOrder.js:8]: you should add some validation using Joi. take a look at this for reference 

https://github.com/saifulislamdev/sylica-backend/blob/main/src/routes/controllers/productsControllers/createProduct.js

File: src/routes/controllers/ordersController/createOrder.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import order from '../../../models/order';
+import OrderModel from '../../../models/order';
+import Joi from '@hapi/joi';
+import validateData from '../../../helpers/validateData';
+
+const AddressValidation = Joi.object({
```
Comment (zealptl) [inline src/routes/controllers/ordersController/createOrder.js:6]: use camelCase everywhere, i.e, addressValidation

File: src/routes/controllers/ordersController/recentOrder.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import OrderModel from '../../../models/order';
+
+const recentOrder = async (req, res) => {
```
Comment (zealptl) [inline src/routes/controllers/ordersController/recentOrder.js:3]: rename to getRecentOrders for consistency

File: src/routes/controllers/ordersController/recentOrder.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  try {
+    //getting user id from request body
+    const user = req.user.id;
+    //get orders from database for the user id
+    const orders = await OrderModel.find({ user });
+    //send response
+    res.status(200).json(orders);
```
Comment (zealptl) [inline src/routes/controllers/ordersController/recentOrder.js:10]: .json({orders}) for consistency

File: package.json (hunk 2 above)
Comment (Mhshohag2256) [reply package.json:47]: It gives me error and the id verification doesn't work. So I quick fix it through vs and it installed automatically.

File: src/routes/controllers/ordersController/createOrder.js (hunk 4 above)
Comment (Mhshohag2256) [reply src/routes/controllers/ordersController/createOrder.js:6]: ok

File: src/routes/controllers/ordersController/recentOrder.js (hunk 5 above)
Comment (Mhshohag2256) [reply src/routes/controllers/ordersController/recentOrder.js:3]: all right

File: src/routes/controllers/ordersController/recentOrder.js (hunk 6 above)
Comment (Mhshohag2256) [reply src/routes/controllers/ordersController/recentOrder.js:10]: added.

**Review by zealptl — APPROVED** [review]

**Pull request merged**